tensorflow/lite/kernels/eigen_support.cc

```diff
@@ -68,7 +68,8 @@ RefCountedEigenContext* GetEigenContext(TfLiteContext* context) {
 void InitDevice(TfLiteContext* context, RefCountedEigenContext* ptr) {
   int num_threads = 4;
   if (context->recommended_num_threads != -1) {
-    num_threads = context->recommended_num_threads;
+    num_threads = (context->recommended_num_threads == 1) ? 0 :
```

Comment:
Just confirming that a thread count of 0 doesn't cause any issues in the thread pool? I'm not sure this is a well-tested path in Eigen? An alternative would be to simply reset the thread pool and device in this case, and clients would know not to use this when the thread count is 1?

Reply:
I doubt this makes much of a difference at the moment, since conv2d is the only one using threads, and when recommended_num_threads==1 it will choose a path that is independent of eigen. In principle the PR looks right, but I second the question about the thread-pool working as expected for  num_threads==0.